hspec-discover/src/Test/Hspec/Discover/Run.hs

```diff
@@ -96,7 +96,7 @@ moduleNameFromId = reverse . dropWhile (== '.') . dropWhile (/= '.') . reverse
 
 -- | Generate imports for a list of specs.
 importList :: Maybe [Spec] -> ShowS
-importList = foldr (.) "" . map f . maybe [] moduleNames
+importList = foldr ((.) . f) "" . maybe [] moduleNames
```

Comment:
I prefer the old version. Yes, it may be less efficient (probably it's not, fusion should take care of it), but I think it more closely expresses intention. We are doing two things here, first map over the list then concatenate.

If anything, we would want to assign `concat_ = foldr (.) ""` and use that, or better yet newtype-wrap `ShowS` and use `mconcat`. But that would be a separate PR.

So for now let's keep the existing code.

Reply:
Ok. Will do.